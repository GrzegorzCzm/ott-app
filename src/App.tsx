import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { useSelector } from "react-redux";

// import "./App.css";
import Player from "./components/player/Player";
import Main from "./components/main/Main";
import Login from "./components/login/Login";
import Alert from "./components/alert/Alert";

interface RenderActionProps {
  location: string;
}

const PrivateRoute = ({ children, userAuthenticated, ...rest }: any) => {
  const render = ({ location }: RenderActionProps) =>
    userAuthenticated ? (
      children
    ) : (
      <Redirect
        to={{
          pathname: "/login",
          state: { from: location },
        }}
      />
    );

  return <Route {...rest} render={render} />;
};

const App = () => {
  const { userAuthenticated } = useSelector((state: any) => state.auth);

  return (
    <Router>
      <Alert />
      <div>
        <Switch>
          <Route path="/login">
            <Login />
          </Route>
          <PrivateRoute
            path="/player/:id"
            userAuthenticated={userAuthenticated}
          >
            <Player />
          </PrivateRoute>
          <PrivateRoute path="/" userAuthenticated={userAuthenticated}>
            <Main />
          </PrivateRoute>
        </Switch>
      </div>
    </Router>
  );
};

export default App;
