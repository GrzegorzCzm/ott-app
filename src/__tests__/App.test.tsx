import React from "react";
import * as redux from "react-redux";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "../App";

describe("App", () => {
  const spy = jest.spyOn(redux, "useSelector");
  spy.mockReturnValue({ userAuthenticated: false });

  it("renders correctly", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
