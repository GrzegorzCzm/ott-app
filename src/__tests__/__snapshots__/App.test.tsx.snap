// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <Alert />
  <div>
    <Switch>
      <Route
        path="/login"
      >
        <Login />
      </Route>
      <PrivateRoute
        path="/player/:id"
        userAuthenticated={false}
      >
        <WithWidth(Player) />
      </PrivateRoute>
      <PrivateRoute
        path="/"
        userAuthenticated={false}
      >
        <SpacingGrid />
      </PrivateRoute>
    </Switch>
  </div>
</BrowserRouter>
`;
